### PR TITLE
Removal of fakeClient from etcd_watcher_test

### PR DESCRIPTION
**Pull request by timothysc**
Continuation from #15349, stemming from #11962
This commit removes fakeClient from etcd_helper_watcher.go in lieu of NewEtcdTestClientServer.

This one actually uncovered a bug in the old client. 

@lavalamp @wojtek-t

### CODE REVIEW

**Comment by timothysc** [comment]
it's a flake test, but I'll wait on @wojtek-t review prior to re-kick.

File: pkg/storage/etcd/etcd_watcher_test.go [hunk 1]
```diff
@@ -320,405 +283,163 @@ func makeSubsets(ip string, port int) []api.EndpointSubset {
...
 
-		fakeClient.WaitForWatchCompletion()
-		if e, a := testCase.Response.R.EtcdIndex+1, fakeClient.WatchIndex; e != a {
-			t.Errorf("%s: expected watch index to be %d, got %d", k, e, a)
-		}
+	// set before the watch and verify events
+	err := h.Set(context.TODO(), key, pod, pod, 0)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
```
Comment (wojtek-t) [inline pkg/storage/etcd/etcd_watcher_test.go]: Can you please add CAS operation here too? I would like to verify two operations touching the same object that happened before are concatenated (I believe they should be, right?)

File: pkg/storage/etcd/etcd_watcher_test.go [hunk 2]
```diff
@@ -320,405 +283,163 @@ func makeSubsets(ip string, port int) []api.EndpointSubset {
...
-		R: &etcd.Response{
-			Node: nil,
-		},
-		E: &etcd.EtcdError{
-			Index:     2,
-			ErrorCode: 100,
-		},
-	}
-	h := newEtcdHelper(fakeClient, codec, etcdtest.PathPrefix())
+func TestWatchListIgnoresRootKey(t *testing.T) {
```
Comment (wojtek-t) [inline pkg/storage/etcd/etcd_watcher_test.go]: Why is it called "IgnoresRootKey"?
It seems that nothing is ignored in this test...

**Comment by wojtek-t** [comment]
I've added two relatively minor comments - other than that LGTM

File: pkg/storage/etcd/etcd_watcher_test.go (hunk 1 above)
Comment (timothysc) [inline pkg/storage/etcd/etcd_watcher_test.go]: Done.

File: pkg/storage/etcd/etcd_watcher_test.go (hunk 2 above)
Comment (timothysc) [inline pkg/storage/etcd/etcd_watcher_test.go]: B/c it ignores the changes to the rootkey it only notices changes to children of the rootkey.  

The select statement below is the validation that changes to the rootkey itself are ignored.

**Comment by timothysc** [comment]
@wojtek-t comments addressed PTAL.

File: pkg/storage/etcd/etcd_watcher_test.go (hunk 2 above)
Comment (wojtek-t) [inline pkg/storage/etcd/etcd_watcher_test.go]: Got it - thanks.

**Comment by wojtek-t** [comment]
lgtm

**Pull request merged**